ice-adapter/src/main/java/com/faforever/iceadapter/util/PingWrapper.java

```diff
@@ -0,0 +1,63 @@
+package com.faforever.iceadapter.util;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import com.google.common.io.CharStreams;
+import org.apache.commons.lang3.SystemUtils;
+
+
+/*
+ * A wrapper around calling the system `ping` executable to query the latency of a host.
+ */
+public class PingWrapper {
+    static final Pattern WINDOWS_OUTPUT_PATTERN = Pattern.compile("Average = (\\d+)ms", Pattern.MULTILINE);
+    static final Pattern UNIX_OUTPUT_PATTERN = Pattern.compile("min/avg/max/mdev = [0-9.]+/([0-9.]+)/[0-9.]+/[0-9.]+", Pattern.MULTILINE);
+
+    /*
+     * Get the round trip time to an address.
+     */
+    public static CompletableFuture<Double> getRTT(String address) {
+        try {
+            Process process;
+            Pattern output_pattern;
+
+            if (SystemUtils.IS_OS_WINDOWS) {
```

Comment:
I don't think this justifies adding another library.
```System.getProperty("os.name").startsWith("Windows");```

Reply:
From what I read `System.getProperty` is buggy and several examples I saw used this library.

Reply:
https://github.com/apache/commons-lang/blob/fe95edd45ffe8d4ffcec17cd8d7741687374c17f/src/main/java/org/apache/commons/lang3/SystemUtils.java#L712

Apache commons uses the same statement for obtaining the system it is running on and then compares it using startsWith("Windows").